frontend/src/components/comparison-page/comparison-page.tsx

```diff
@@ -8,10 +8,38 @@ import { CompTopTableBar } from '@components/comp-top-table-bar/comp-top-table-b
 import { OptionsSubtable } from '@components/comparison-options-table/options-subtable';
 import { GeneralComparisonTable } from '@components/general-comparison-table/general-comparison-table';
 import { Header } from '@components/header/header';
+import { useGetComparisonGeneralInfoQuery } from '@store/queries/comparisons';
 
 import styles from './styles.module.scss';
 
 const ComparisonPage: React.FC = () => {
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const [isOnlyDifferenceShown, setIsOnlyDifferenceShown] = useState(true); // this will be used while implementing toggle button
+
+  const { data, isLoading } = useGetComparisonGeneralInfoQuery();
+
+  const showOptionsTables = (): (React.ReactElement | undefined)[] => {
+    if (isOnlyDifferenceShown) {
+      return useMemo(
+        () =>
+          complectationOptions.map((option) => {
+            if (data?.some((car) => car.options[option].length)) {
+              return <OptionsSubtable key={option} title={option} />;
+            }
+          }),
+        [data],
+      );
+    }
+
+    return useMemo(
+      () =>
+        complectationOptions.map((option) => (
+          <OptionsSubtable key={option} title={option} />
+        )),
+      [data],
+    );
+  };
```

Comment:
It would be better to extract these 2 `useMemo` into 2 sub-components. It would be a much more idiomatic React solution

Reply:
Do we need to wrap there sub-components in `React.memo()` or...
Why is it better? Won't they re-render each time user toggles table view?